src/collective/jsonmigrator/blueprints/datafields.py

```diff
@@ -27,8 +29,15 @@ def __init__(self, transmogrifier, name, options, previous):
         self.datafield_prefix = options.get("datafield-prefix", "_datafield_")
         self.root_path_length = len(self.context.getPhysicalPath())
 
+        if "path-key" in options:
+            pathkeys = options["path-key"].splitlines()
+        else:
+            pathkeys = defaultKeys(options["blueprint"], name, "path")
+        self.pathkey = Matcher(*pathkeys)
+
     def __iter__(self):
         for item in self.previous:
+            pathkey = self.pathkey(*list(item.keys()))[0]
```

Comment:
Same here: Was this supposed to go in here? This does not have anything to do with code analysis.

Reply:
Flake8 complained that the `pathkey` variable was not defined. And it really wasn't:

https://github.com/collective/collective.jsonmigrator/blob/32b58f717e7f01356d9d1e19566c4b2d6cc9ffce/src/collective/jsonmigrator/blueprints/datafields.py#L30-L38

This error was caused by the commit:

https://github.com/collective/collective.jsonmigrator/commit/32b58f717e7f01356d9d1e19566c4b2d6cc9ffce#diff-12d065283fd4ff3aa1d267db4a49c4c83f62d6b841979d9fe6ed11e8256ddeebL38

All blueprints define the `pathkey` variable, except this one.

The definition of this variable, enables to define a custom key in json, to specify the object path. So I decided to add this functionality to this bluiprint as well.

I am not using this blueprint. That's why I didn't notice this error before.

Reply:
@mauritsvanrees I decided to fix in this PR only the undefined variable. We can add pathkey support in another PR.

Can you please review it again?

Reply:
LGTM